nav2_route/test/test_geojson_graph_file_loader.cpp

```diff
@@ -0,0 +1,252 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <nlohmann/json.hpp>
+
+#include "nav2_route/plugins/graph_file_loaders/geojson_graph_file_loader.hpp"
+#include "nav2_route/graph_loader.hpp"
+
+class RclCppFixture
+{
+public:
+  RclCppFixture() {rclcpp::init(0, nullptr);}
+  ~RclCppFixture() {rclcpp::shutdown();}
+};
+RclCppFixture g_rclcppfixture;
+
+using namespace nav2_route; // NOLINT
+using Json = nlohmann::json;
+
+TEST(GeoJsonGraphFileLoader, file_does_not_exist)
+{
+  Graph graph;
+  GraphToIDMap graph_to_id_map;
+  std::string file_path = "graph.geojson";
+
+  GeoJsonGraphFileLoader graph_file_loader;
+  bool result = graph_file_loader.loadGraphFromFile(graph, graph_to_id_map, file_path);
+  EXPECT_FALSE(result);
+}
+
+TEST(GeoJsonGraphFileLoader, no_nodes_in_graph)
+{
+  Json json_obj = nlohmann::json::parse(
+    R"(
+  {
+    "features": [
+    { "type": "Feature",
+      "properties":
+      {
+        "id": 123,
+        "startid": 20,
+        "endid": 4
+      },
+      "geometry":
+      {
+        "type": "MultiLineString",
+        "coordinates": [ [ [ 1.0, 3.65 ], [ -4.0, 3.65 ] ] ]
+      }
+    }
+  ]
+  }
+  )");
+  std::string file_path = "no_nodes.geojson";
+
+  std::ofstream missing_id_file(file_path);
+  missing_id_file << json_obj;
+  missing_id_file.close();
+
+  Graph graph;
+  GraphToIDMap graph_to_id_map;
+  GeoJsonGraphFileLoader graph_file_loader;
+  bool result = graph_file_loader.loadGraphFromFile(graph, graph_to_id_map, file_path);
+  EXPECT_FALSE(result);
+  std::filesystem::remove(file_path);
+}
+
+TEST(GeoJsonGraphFileLoader, no_edges_in_graph)
+{
+  Json json_obj = nlohmann::json::parse(
+    R"(
+  {
+    "features": [
+    { "type": "Feature",
+      "properties":
+      {
+        "frame": "map"
+      },
+      "geometry":
+      {
+        "type": "Point",
+        "coordinates": [ 0.0, 0.0 ]
+      }
+    }
+  ]
+  }
+  )");
+  std::string file_path = "no_edges.geojson";
+
+  std::ofstream missing_id_file(file_path);
+  missing_id_file << json_obj;
+  missing_id_file.close();
+
+  Graph graph;
+  GraphToIDMap graph_to_id_map;
+  GeoJsonGraphFileLoader graph_file_loader;
+  bool result = graph_file_loader.loadGraphFromFile(graph, graph_to_id_map, file_path);
+  EXPECT_FALSE(result);
+  std::filesystem::remove(file_path);
+}
+
+TEST(GeoJsonGraphFileLoader, missing_node_id)
```

Comment:
1 more test: what is a node which the edge is referring to doesn't exist anymore (e.g. due to someone manipulating a graph and messing it up)